test/creator.cpp

```diff
@@ -151,54 +187,117 @@ TEST(ZimCreator, createZim)
+  direntOffset = offset_t(reader->read_uint<offset_type>(offset_t(urlPtrPos + 8*(delta++))));
+  dirent = direntReader.readDirent(direntOffset);
+  test_article_dirent(dirent, 'M', "Title", "Title", plain_mimetype, cluster_index_t(0), Filter<blob_index_t>());
+  auto metaBlobIndex = dirent->getBlobNumber();
+
+  direntOffset = offset_t(reader->read_uint<offset_type>(offset_t(urlPtrPos + 8*(delta++))));
   dirent = direntReader.readDirent(direntOffset);
-  test_article_dirent(dirent, 'M', "Title", "Title", plain_mimetype, cluster_index_t(0), blob_index_t(1));
+  test_redirect_dirent(dirent, 'W', "mainPage", "mainPage", entry_index_t(0));
 
 #if defined(ENABLE_XAPIAN)
-  direntOffset = offset_t(reader->read_uint<offset_type>(offset_t(urlPtrPos + 24)));
+  direntOffset = offset_t(reader->read_uint<offset_type>(offset_t(urlPtrPos + 8*(delta++))));
   dirent = direntReader.readDirent(direntOffset);
   test_article_dirent(dirent, 'X', "fulltext/xapian", "fulltext/xapian", xapian_mimetype, cluster_index_t(1), Filter<blob_index_t>());
+#endif
+
+  direntOffset = offset_t(reader->read_uint<offset_type>(offset_t(urlPtrPos + 8*(delta++))));
+  dirent = direntReader.readDirent(direntOffset);
+  test_article_dirent(dirent, 'X', "listing/titleOrdered/v0", Filter<const std::string>(), listing_mimetype, cluster_index_t(1), Filter<blob_index_t>());
+  auto v0BlobIndex = dirent->getBlobNumber();
 
-  direntOffset = offset_t(reader->read_uint<offset_type>(offset_t(urlPtrPos + 32)));
+  direntOffset = offset_t(reader->read_uint<offset_type>(offset_t(urlPtrPos + 8*(delta++))));
+  dirent = direntReader.readDirent(direntOffset);
+  test_article_dirent(dirent, 'X', "listing/titleOrdered/v1", Filter<const std::string>(), listing_mimetype, cluster_index_t(1), Filter<blob_index_t>());
+  auto v1BlobIndex = dirent->getBlobNumber();
+
+#if defined(ENABLE_XAPIAN)
+  direntOffset = offset_t(reader->read_uint<offset_type>(offset_t(urlPtrPos + 8*(delta++))));
   dirent = direntReader.readDirent(direntOffset);
   test_article_dirent(dirent, 'X', "title/xapian", "title/xapian", xapian_mimetype, cluster_index_t(1), Filter<blob_index_t>());
 #endif
 
   auto clusterPtrPos = header.getClusterPtrPos();
+
+  // Test main content
   auto clusterOffset = offset_t(reader->read_uint<offset_type>(offset_t(clusterPtrPos)));
   auto cluster = Cluster::read(*reader, clusterOffset);
   ASSERT_EQ(cluster->getCompression(), CompressionType::zimcompZstd);
-  ASSERT_EQ(cluster->count(), blob_index_t(2));
-  auto blob = cluster->getBlob(blob_index_t(0));
+  ASSERT_EQ(cluster->count(), blob_index_t(3));
+
+  auto blob = cluster->getBlob(fooBlobIndex);
   ASSERT_EQ(std::string(blob.data(), blob.size()), "FooContent");
-  blob = cluster->getBlob(blob_index_t(1));
+
+  blob = cluster->getBlob(foo2BlobIndex);
+  ASSERT_EQ(std::string(blob.data(), blob.size()), "Foo2Content");
+
+  blob = cluster->getBlob(metaBlobIndex);
   ASSERT_EQ(std::string(blob.data(), blob.size()), "This is a title");
```

Comment:
Too much noise in this code. Introducing a simple helper function `std::string toString(const zim::Blob& )` will enable to have here only assertions of the form

```
ASSERT_EQ(toString(cluster->getBlob(???BlobIndex)), ".....");
```

Reply:
In fact, there is already a `std::string` operator for blob.
Replaced by `ASSERT_EQ(std::string(cluster-getBlob(???BlobIndex)), "...");`